telegram/base.py

```diff
@@ -80,3 +80,25 @@ def to_dict(self):
                     data[key] = value
 
         return data
+
+    def _get_id(self):
+        for x in ['id', 'file_id', 'message_id', 'result_id', 'update_id']:
+            if hasattr(self, x):
+                return self.__getattribute__(x)
+        raise NotImplementedError
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            try:
+                return self._get_id() == other._get_id()
```

Comment:
Nitpick: you're accessing a private method `_get_id()` of something other than `self`.
Maybe it's the proper thing to do here, just wish that we think about it one more time.
Maybe `return hash(self) == hash(other)`?